lib/exercism/user_finished_tracks.rb

```diff
@@ -24,9 +24,19 @@ def started_tracks
     ActiveRecord::Base.connection.execute(completed_count_sql).to_a
   end
 
+  def count_finished_problems(user_finished_exercises, track)
+    (user_finished_exercises & track.problems.map(&:slug)).size
+  end
+
+  def completed?(track_attributes)
```

Comment:
Usually when you name a method with `?` it is expected that the method will return a boolean. In this can it could be `problems_solved >= track.problems.count`.  I think it was my fault since I suggested that we could remove the `track` variable from the `completed_tracks` method, which is not true at all. What do you think of:

```ruby
def completed_tracks
   track = tracks.find { |t| t.id == db_row['track_id'] }
   arr << track if completed?(db_row, track)
   arr
end

def completed?(user_track, full_track)
  problems_solved = count_finished_problems(user_track['completed_problems'].split(','), full_track)
  problems_solved >= full_track.problems.size
end
```

Sorry for the trouble @MohanL, let me know if you have time to change this.